### DISCUSSION TITLE
Text indentations aren't converted properly

### DISCUSSION THREAD

**Discussion by sojusnik** [peter88213/zim2obsidian#15] [Ideas]
Indentations of text by using the tab key aren't converted properly after Zim's markdown export, although indentations with checkboxes do work, see examples below.

**To Reproduce**
1. Copy the following to your zim file:
```
indentations by using tab aren't converted properly after Zim's markdown conversion
	indentation of normal text by using one tab
		indentation of normal text by using two tabs
			indentation of normal text by using three tabs

[ ] indentations with checkboxes work though
	[ ] indented checkbox by using one tab
		[ ] indented checkbox by using two tabs

[ ] indentations following a checkbox aren't converted properly either
	indentation of normal text by using one tab
		indentation of normal text by using two tabs
			indentation of normal text by using three tabs
```
2. Convert this page via Zim's export function to markdown
3. Open in Obsidian

**Expected behavior**
Text should be indented, as it is displayed in Zim.

**Comment by peter88213** [comment]
When pressing the "Tab" key with Zim, a tab character (U+0009) is inserted.
Be aware that tabs, like multiple blanks, aren't specified in Markdown. It's advised to use  `&nbsp;` or `&emsp;` instead when typing Markdown formatted text with a text editor. 

However, the tab character gets lost upon Zim's Markdown export (see [open issues here](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/develop/zim/formats/markdown.py)). This is why *zim2obsidian* cannot keep it using the method mentioned above.

**Comment by peter88213** [comment]
A solution is now proposed here:
https://github.com/peter88213/zim2obsidian/issues/13#issuecomment-1650446635

**Reply by sojusnik** [reply]
I'm continuing here to have a better separation from the other issue regarding checkboxes.

> What I could imagine would be a "dirty hack". One could edit the original Zim notebook with a preprocessor and replace all tab characters with a special code. Then export with _Zim_ and post-process with _zim2obsidian_. Then at the very end, have the special codes in the Markdown files replaced with tab characters again.

This is epic! Thank you very much!

I've followed through your steps and one issue arises: Indented checkboxes were already converted fine before your solution with _subst_indent.py_ and _indent_md.py_, but now they aren't rendered properly.

This is how the source in Zim looks like:
```
[ ] indentations with checkboxes
	[ ] indented checkbox by using one tab
		[ ] indented checkbox by using two tabs
```

Then after _zim2obsidian.py_ it already looks perfect (source in Obsidian):

```
- [ ] indentations with checkboxes
	- [ ] indented checkbox by using one tab
		- [ ] indented checkbox by using two tabs
```

But if _subst_indent.py_ and _indent_md.py_ and afterwards _zim2obsidian.py_ are applied then it looks like (source in Obsidian):

```
- [ ] indentations with checkboxes work though

	[ ] indented checkbox by using one tab
		[ ] indented checkbox by using two tabs
```

Keep in mind that the new line after the checkbox is applied due to the behaviour described [here](https://github.com/peter88213/zim2obsidian/issues/13#issuecomment-1652425733).

Normal indented text is converted perfectly fine though.

**Reply by peter88213** [reply]
See: https://github.com/peter88213/zim2obsidian/issues/13#issuecomment-1652436265

**Reply by peter88213** [reply]
See also: https://github.com/peter88213/zim2obsidian/issues/20#issuecomment-2020645512

**Comment by peter88213** [comment]
With version 0.13.0, the *zim2obsidian.py* script converts by default blocks that are indented with tabs to Obsidian code blocks that are fenced with three backticks. This corresponds to the *Zim* specification for verbatim text export.

With the command line parameter `--backticks` this behavior is overridden. Then *zim2obsidian.py* keeps backticks-enclosed passages verbatim and passes leading tabs through without adding code block markers. This corresponds to the behaviour requested with #21 and #22.
